core/src/main/java/org/apache/accumulo/core/clientImpl/ConditionalWriterImpl.java

```diff
@@ -679,7 +679,7 @@ private void invalidateSession(SessionID sessionId, HostAndPort location)
 
     LockID lid = new LockID(context.getZooKeeperRoot() + Constants.ZTSERVERS, sessionId.lockId);
 
-    while (true) {
+    while (true && !Thread.currentThread().isInterrupted()) {
```

Comment:
I think that prior to this change this loop would exit either when invalidation was successful or if an error was thrown. I think this introduces a new condition where the loop exits but it's not successful nor is an error thrown.  I'm assuming here (because I didn't look) that the calling code will assume success when this returns, even though it may not have succeeded. This would be a good place to throw InterruptedException if we want to handle it.

Reply:
See 60f14d86ef8b - the other option would be to remove the interrupt flag check and ignore as done before.